src/saml2/client_base.py

```diff
@@ -183,6 +183,10 @@ def __init__(self, config=None, identity_cache=None, state_cache=None,
 
             setattr(self, attr, val)
 
+        # signing and digest algs
+        self.signing_algorithm = self.config.getattr('signing_algorithm', "sp")
```

Comment:
what will happen if no `signing_algorithm` or `digest_algorithm` is set? These will be set to `None`, right? And will that work or will it fail? Maybe we want to have a default value for these two if nothing is set? If we want to have a default value, it should be set in the `attribute_defaults` which is few lines above this line

Reply:
I though on this aspect and as it is at this moment it would rely on None and this will use the default "hardcoded" hash.
I agree that it would be better to deal with a pythonic-get-default, exactly there

Reply:
Hi @ioparaskev, what do you think about something like this:
https://github.com/IdentityPython/pysaml2/issues/738#issuecomment-717340111